### PR TITLE
Incremental scanning progress in UI

### PR DESCRIPTION
**Pull request by jeske**
TLDR - This causes scanned image data to appear incrementally, instead of only once the scan is done.

This adds a new DataSourceManager.UseIncrementalMemoryXfer boolean flag (default false), which when set to true causes TwainDotNet to use new code for incremental TWAIN ImageMemXfer, instead of the existing blocking ImageNativeXfer.  

Currently the new Incremental mode only supports TWAIN 1 bpp and 24 bpp RGB image data. It does not support 8bpp greyscale,  40bpp CMYK, or Planar RRR-GGG-BBB formats, because these formats are not supported by Gdi32Native.SetDIBitsToDevice. These formats could either be supported via other conversion means, or the code could be changed to fallback to ImageNativeXfer for unsupported formats.

This also fixes a memory leak in DataSourcemanager.TransferPictures(), where hbitmap is not deallocated if a TWAIN failure code causes an early-exit before the IDisposable BitmapRenderer was created to manage it's lifetime. The hbitmap to Bitmap conversion is now a static method, and TransferPictures() closes lifetime on hbitmap in a finally clause. 

It also adds a TwainBool enum, because this added tests for twain boolean values, and it felt a bit error prone to just use "1" and "0" numbers.

### CODE REVIEW

**Comment by jeske** [comment]
updated to support 1-bit and 8-bit greyscale via MemoryXfer
